src/kbc_modules/offline_fs_kbc/mod.rs

```diff
@@ -102,111 +104,82 @@ mod tests {
-            name: ResourceName::Policy.to_string(),
-            optional: HashMap::new(),
-        })
-        .unwrap();
-        assert_eq!(
-            kbc.get_resource(&policy_rd).await.unwrap(),
-            POLICYJSON.as_bytes()
-        );
-
-        let sigstore_config_rd = serde_json::to_string(&ResourceDescription {
-            name: ResourceName::SigstoreConfig.to_string(),
-            optional: HashMap::new(),
-        })
-        .unwrap();
-        assert_eq!(
-            kbc.get_resource(&sigstore_config_rd).await.unwrap(),
-            SIGSTORECONFIG.as_bytes()
-        );
-
-        let public_key_rd = serde_json::to_string(&ResourceDescription {
-            name: ResourceName::GPGPublicKey.to_string(),
-            optional: HashMap::new(),
-        })
-        .unwrap();
-        assert_eq!(
-            kbc.get_resource(&public_key_rd).await.unwrap(),
-            PUBKEY.as_bytes()
-        );
-
-        let cosign_key_rd = serde_json::to_string(&ResourceDescription {
-            name: ResourceName::CosignVerificationKey.to_string(),
-            optional: HashMap::new(),
-        })
-        .unwrap();
-        assert_eq!(
-            kbc.get_resource(&cosign_key_rd).await.unwrap(),
-            COSIGNKEY.as_bytes()
-        );
-
-        let credential_rd = serde_json::to_string(&ResourceDescription {
-            name: ResourceName::Credential.to_string(),
-            optional: HashMap::new(),
-        })
-        .unwrap();
-        assert_eq!(
-            kbc.get_resource(&credential_rd).await.unwrap(),
-            CREDENTIAL.as_bytes()
-        );
-
-        // Case 2. Error while get bad resource name from a good kbc instance
-        assert!(kbc.get_resource("bad").await.is_err());
-
-        // Case 3. Error while get good resource name from bad kbc instance
-        let mut resources_load_failure_kbc = OfflineFsKbc {
-            kbs_info: HashMap::new(),
-            keys: Ok([(KID.to_string(), KEY.to_vec())].iter().cloned().collect()),
-            resources: Err(anyhow!("")),
-        };
-        let good_policy_rd = serde_json::to_string(&ResourceDescription {
-            name: ResourceName::Policy.to_string(),
-            optional: HashMap::new(),
-        })
-        .unwrap();
+        }
+    }
 
-        assert!(resources_load_failure_kbc
-            .get_resource(&good_policy_rd)
-            .await
-            .is_err());
+    #[rstest::rstest]
+    // Case 1. Get resources from good kbc instance correctly
+    #[case(true, ResourcePath::Policy.as_ref(), POLICYJSON)]
+    #[case(true, ResourcePath::SigstoreConfig.as_ref(), SIGSTORECONFIG)]
+    #[case(true, ResourcePath::GPGPublicKey.as_ref(), PUBKEY)]
+    #[case(true, ResourcePath::CosignVerificationKey.as_ref(), COSIGNKEY)]
+    #[case(true, ResourcePath::Credential.as_ref(), CREDENTIAL)]
+    // Case 2. Error while get bad resource name from a good kbc instance
+    #[case(false, "bad", "")]
```

Comment:
I meet a test error at here, maybe the bad cases need be updated, would you like do a check?
```bash
root@peer-pod1:~/attestation-agent# make KBC=offline_fs_kbc && make install
root@peer-pod1:~/attestation-agent# cargo test --features offline_fs_kbc,rust-crypto
warning: skipping duplicate package `app` found at `/root/.cargo/git/checkouts/sgxdatacenterattestationprimitives-d6934a418e6beae0/85cf8bd/SampleCode/RustTDQuoteGenerationSample`
    Finished test [unoptimized + debuginfo] target(s) in 0.14s
     Running unittests src/lib.rs (target/debug/deps/attestation_agent-c31f2de2717cdaeb)

running 12 tests
test kbc_modules::offline_fs_kbc::tests::test_get_key ... ok
test kbc_modules::offline_fs_kbc::tests::test_get_resource::case_1 ... ok
test kbc_modules::offline_fs_kbc::tests::test_get_resource::case_2 ... ok
test kbc_modules::offline_fs_kbc::tests::test_get_resource::case_3 ... ok
test kbc_modules::offline_fs_kbc::tests::test_get_resource::case_4 ... ok
test kbc_modules::offline_fs_kbc::tests::test_get_resource::case_5 ... ok
test kbc_modules::offline_fs_kbc::tests::test_get_resource::case_6 ... FAILED
test kbc_modules::uri::tests::conversions ... ok
test kbc_modules::uri::tests::serialize ... ok
test kbc_modules::uri::tests::deserialize ... ok
test kbc_modules::offline_fs_kbc::common::tests::test_load_keys ... ok
test kbc_modules::offline_fs_kbc::common::tests::test_load_resources ... ok

failures:

---- kbc_modules::offline_fs_kbc::tests::test_get_resource::case_6 stdout ----
thread 'kbc_modules::offline_fs_kbc::tests::test_get_resource::case_6' panicked at 'called `Result::unwrap()` on an `Err` value: "invalid kbs uri, should be kbs://<addr-of-kbs>/<repo>/<type>/<tag>"', src/kbc_modules/offline_fs_kbc/mod.rs:175:53
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    kbc_modules::offline_fs_kbc::tests::test_get_resource::case_6

test result: FAILED. 11 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s
```

Reply:
Thanks for pointing out! This is caused by the KBS URI parsed error. I've fixed this. There are also no related CI to cover. I've updated `basic.yaml` CI to cover this.